Documentation, some minimal refactoring.

Jukelyn opened the pull request:
Let me know what you think! Also I saw that you commented
```python
        # Maybe I'll implement this at some point. idk
```
on line 140 of `led_matrix.py`. What were wanting to implement? I can take a shot at it if you'd like!

Jukelyn committed:
Added requirements.txt

Jukelyn committed:
Merge branch 'main' of https://github.com/Jukelyn/led_matrix.py

Jukelyn committed:
Added gitignore, added documentation and renamed some ambiguous variables, lastly changed quit() to sys.exit() per Pylint's R1722.

Jukelyn reviewed (commented):
I forgot to mention but I also changed the general Exceptions to more purposeful ValueErrors instead.

Jukelyn commented:
Also, you should consider adding a license file. See [here](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/licensing-a-repository)

Encampeded commented:
Not sure how necessary the multi-line comments explaining each function are—I'd prefer that sort of documentation stay in the readme—but otherwise looks great. I might revert some stylistic stuff just to my taste (one-line if statements, line breaks) when I get around to it, but the type hints and other edits look good.

The ```# Maybe I'll implement this``` was in regards to diagonal line drawing in draw_line(). I had it working in an older version, but it never looked very good and I left it out when I recently rewrote most of this. Frankly, from what I've seen, diagonal lines look pretty bad on this thing anyways, and I really don't see a good use for them. I think I'll leave it out for now for the sake of simplicity

I'll look into adding a license. Thanks for the suggestion(s), and your time!

Jukelyn commented:
> Not sure how necessary the multi-line comments explaining each function are—I'd prefer that sort of documentation stay in the readme—but otherwise looks great. I might revert some stylistic stuff just to my taste (one-line if statements, line breaks) when I get around to it, but the type hints and other edits look good.

The multiline comments are for documentation, docstrings, they are useful if somebody wants to use `__doc__` or `help()` and also if you want to use something like mkdocs to generate a site that provides a clean format of your code's documentation so that others can easily view each method and understand what all is going on. It is optional but I find it to be helpful as practice and habit. 

> I'll look into adding a license. Thanks for the suggestion(s), and your time!

No problem, just trying to find some things to contribute to here and there!

Encampeded commented:
Maybe for a larger project, but for something this size, I think most functions are pretty self documenting and I've left comments when they're not

Jukelyn commented:
Fair enough!

Encampeded committed:
revised formatting

- Removed multi-line comments describing every function/method
- Fixed a couple incorrect/missing type hints
- Reverted some other general formatting to taste
